.github/workflows/build.yml

```diff
@@ -10,16 +10,15 @@ jobs:
 
     steps:
       - name: 'Checkout source code'
-        uses: actions/checkout@v2
-
+        uses: actions/checkout@v4
       - name: 'Set up JDK'
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v3
         with:
           java-version: '11'
           distribution: 'adopt'
 
       - name: 'Cache dependencies'
-        uses: actions/cache@v2
+        uses: actions/cache@v3
```

Comment:
I think it would be a good idea to ditch the cache action and just use the [cache](https://github.com/actions/setup-java#caching-maven-dependencies) functionality in `setup-java` instead. Should be as easy as adding `cache: 'maven'`.